vllm/v1/sample/logits_processor/__init__.py

```diff
@@ -185,6 +191,17 @@ def build_logitsprocs(
             " do not support logits processors."
         )
         return LogitsProcessors()
+
+    # Check if speculative decoding is enabled.
+    if vllm_config.speculative_config:
+        if custom_logitsprocs:
+            raise ValueError(STR_SPEC_DEC_REJECTS_LOGITSPROCS)
+        logger.debug(
```

Comment:
I just realized that this means the built-in logits processors also won't work which is a bit unfortunate since from the outside these are no different to other sampling params (penalties etc.)

The plan was/is to port most of the sampling param impls to be LPs once we've stabilized that interface, we had just done this with a couple of them initially in preparation for that. So there will still be some "silent failure" if these parameters are used (min_p, logit_bias, min_tokens), but I guess it's no worse than the current state so probably still ok to get this in and address as part of the larger refactoring underway.

Reply:
Thanks! keep in mind it may look a bit different given the other in-flight/planned rework. For example I expect we'll remove various python lists from the input batch and instead use numpy arrays.